AK/Format.h

```diff
@@ -202,14 +207,8 @@ class FormatBuilder {
         size_t width,
         char fill = ' ');
 
-    const StringBuilder& builder() const
-    {
-        return m_builder;
-    }
-    StringBuilder& builder() { return m_builder; }
-
 private:
-    StringBuilder& m_builder;
+    Function<void(char)> m_putch;
```

Comment:
I will re-raise my concerns about `Function` allocating stuff here (depending on the passed function, this may allocate - currently all the passed functions are safe since they only contain one pointer-sized member), and suggest (again) that we add a `bool may_allocate` template parameter to `Function`, defaulted to `true` which would entirely remove the concept of outline buffer from it.
That makes this path not allocate, and makes the compiler check it for us.

Reply:
I don't think outright forbidding the use of a large functor in `FunctionBuilder` is the right solution. The whole point of being able to take an arbitrary callable object is that it's more flexible. I'm inclined to just add a comment reminding us to only pass in a function pointer, or a `static_assert(IsFunctionPointer<decltype(&critical_console_out)>)`

Reply:
Reminding people is never enough, we'll just end up back in square zero.
We can lift the restriction if a user of the API shows up that has a convincing reason to have the formatting stuff allocate, until then it's better to be safe imo.

Reply:
Yea, it seems like a giant foot gun that to allow this to allocate in kernel mode.

Reply:
> that we add a bool may_allocate template parameter to Function, defaulted to true

Is that even possible? Template parameter packs must be last, so we can't add `may_allocate` after the argument types, and similarly, nothing can come after defaulted template parameters. Moving our current implementation into `AK::Detail` isn't possible either, as template alias declarations can't be partially specialized.